src/components/send-button/index.js

```diff
@@ -195,7 +198,11 @@ export default compose( [
 						) : null }
 						<Button
 							isPrimary
-							disabled={ newsletterValidationErrors.length > 0 }
+							disabled={
+								newsletterValidationErrors.length > 0 ||
+								( window.newspack_newsletters_data.service_provider === 'campaign_monitor' &&
```

Comment:
I now remember why I didn't originally do what you suggested in c6e1df0—for some reason, after saving, `newsletterData` briefly becomes empty during the exact time we need to read it for pre-send checks. You can verify by adding `console.log( newsletterData )` to `send-button/index.js` and watching the console as you click the "Send" button.

This causes the pre-send validation to not work for Campaign Monitor. Also, because the Campaign Monitor validation happens a bit differently (based on the presence of metadata, rather than synced campaign data), the same checks for `newsletterValidationErrors` aren't needed here.

This works for me, even though it's not quite as clean as keeping everything in the ESP files: 081865d

Happy to refactor if you can help me figure out why `newsletterData` keeps disappearing?

Reply:
The disappearance happened because meta fields in the editor were updated on post saving. Since these two meta fields were not registered, they were missing in the response. 
I've registered them in cb4134207a276b8f01b9193795560f479aeaba2b
Let me know if that solves the validation issue.

Also, I noticed that all the meta fields are public – retrievable via the REST API. To prevent that, I've also updated all existing meta fields to be be available only in the editor.

Reply:
This makes a lot of sense to me! Thanks for explaining. I did notice that the "send" button became disabled after saving because the the `newsletterData.campaign` value kept getting lost. I think d35c662 fixes it.